feedly/lib/feedly/api_client/session.py

```diff
@@ -0,0 +1,336 @@
+        self,
+        token_dir: Path = Path.home() / ".config/cte_feedly",
+        client_id: str = "feedlydev",
+        client_secret: str = "feedlydev",
+    ):
+        """
+
+        :param token_dir: the directory to store the tokens
+        :param client_id: the client id to use when refreshing the auth token. the default value works for developer tokens.
+        :param client_secret: the client secret to use when refreshing the auth token. the default value works for developer tokens.
+        """
+        super().__init__(client_id, client_secret)
+        if not token_dir.is_dir():
+            raise ValueError(f"{token_dir.absolute()} does not exist!")
+
+        refresh_path = token_dir / "refresh.token"
+        if refresh_path.is_file():
+            self.refresh_token = refresh_path.read_text().strip()
+
+        self.auth_token_path: Path = token_dir / "access.token"
+        self._auth_token = self.auth_token_path.read_text().strip()
+
+    @Auth.auth_token.setter
+    def auth_token(self, token: str):
+        self._auth_token = token
+        self.auth_token_path.write_text(token)
+
+
+class FeedlySession(APIClient):
+    def __init__(
+        self,
+        auth: Optional[Union[str, Auth]] = None,
+        api_host: str = "https://feedly.com",
+        user_id: str = None,
+        client_name="cte_feedly.python.client",
+        ssl_verification=True,
+        proxy={},
+    ):
+        """
+        :param auth: either the access token str to use when making requests or an Auth object to manage tokens. If none
+         are passed, it is assumed that the token and refresh token are correcly setup in the `~/.config/cte_feedly`
+         directory. You can run setup_auth.py in the examples to get setup.
+        :param api_host: the cte_feedly api server host.
+        :param user_id: the user id to use when making requests. If not set, a request will be made to determine the user from the auth token.
+        :param client_name: the name of your client, set this to something that can identify your app.
+        """
+        super().__init__()
+        if not client_name:
+            raise ValueError("you must identify your client!")
+
+        if isinstance(auth, str):
+            token: str = auth
+            auth = Auth()
+            auth.auth_token = token
+        elif auth is None:
+            auth = FileAuthStore()
+
+        self.auth: Auth = auth
+        self.api_host: str = api_host
+        self.session = Session()
+        self.session.mount(
+            "https://feedly.com", HTTPAdapter(max_retries=1)
+        )  # as to treat cte_feedly server and connection errors identically
+        self.client_name = client_name
+        self.timeout: int = 10
+        self.max_tries: int = 3
+
+        user_data = {"id": user_id} if user_id else {}
+        self._user: FeedlyUser = FeedlyUser(user_data, self)
+        self._valid: bool = None
+        self._last_token_refresh_attempt: float = 0
```

Comment:
:exclamation:**Cycode**: Secret of type: 'Generic Password' was found.
Severity: Critical

Would you like to exclude this secret from your status checks?
Tell us what to do with one of the following hashtags:

| Tag | Short Description |
| --- | --- |
| **#cycode_secret_ignore_here** | Applies to this request only |
| **#cycode_secret_false_positive** | Applies to this secret value for all repos in your organization |

 <!-- f17458c3-2425-42d2-ac76-c70738b9557f -->